src/main/scala/coreplex/Coreplex.scala

```diff
@@ -49,7 +50,9 @@ case class CoreplexConfig(
   val plicKey = PLICConfig(nTiles, hasSupervisor, nExtInterrupts, 0)
 }
 
-abstract class Coreplex(implicit val p: Parameters, implicit val c: CoreplexConfig) extends Module
+abstract class Coreplex(clockSignal: Clock = null, resetSignal: Bool = null)
+    (implicit val p: Parameters, implicit val c: CoreplexConfig)
+    extends Module(Option(clockSignal), Option(resetSignal))
```

Comment:
I tried using `_clock` and `_reset` but even then the line is too long. `implicit` is a very long word. I've changed it for the coreplex classes though.